### DISCUSSION TITLE
How to add a custom mode

### DISCUSSION THREAD

**Discussion by JohanChane** [davatorium/rofi#1301] [Q&A] [answered]
I like the feature like `xfce4-appfinder --collapsed` which i can search the command of starting program in desktop entries after i just install the program. And I have Implemented the feature by write a shell script. 

exec.sh
 
```sh
#!/usr/bin/env bash

if [[ -z "$*" ]];then
    dex  -ad -s /usr/share/applications/:~/.local/share/applications/:/etc/xdg/autostart/:~/.config/autostart/|cut -c `expr length 'Executing command: A'`-
else
    nohup $* >/dev/null 2>&1 &
    exit $?
fi
```

```sh
rofi -show drun -modi drun,run,exec:~/.config/rofi/exec.sh,keys,file-browser,window,ssh
```
I want to add a mode `exec`. and the start commad is:

```sh
rofi -show drun -modi drun,run,exec,keys,file-browser,window,ssh
```
So, how to config ?

**Comment by DaveDavenport** [comment, marked as answer]
I am not sure what the question is, this is correct syntax for 'modi' and will work both on commandline as adding it to config:
```bash
rofi -show drun -modi drun,run,exec:~/.config/rofi/exec.sh,keys,file-browser,window,ssh
```

```css
configuration {
   modi: "drun,run,exec:~/.config/rofi/exec.sh,keys,file-browser,window,ssh";
}
```

**Reply by JohanChane** [reply]
I had an obsession for unified command. Let it goes. I just want to know how to config a custom mode like default modes, which don't need to specify a file in command.

**Reply by DaveDavenport** [reply]
I do not understand the question? 
Your custom mode is a script called by the script mode. You need to tell script mode what script to execute, this you need to do when defining the mode (name:script). There is no other way.
( If the used script is in $PATH, you don't need to specify full path).

For using the script mode, this is the only solution.

**Reply by MarkoPLK** [reply]
You can put your custom script in ~/.config/rofi/scripts and dont need to specify the path of the executable. For example: you have your script in ~/.config/rofi/scripts/exec.sh, so you can call  
`rofi -show drun -modi drun,run,exec,keys,file-browser,window,ssh`

**Reply by DaveDavenport** [reply]
These posts 4 years old, that feature was added a year later.
